--- libsigrok-0.4.0/configure.ac.orig	2016-01-29 23:44:34.000000000 +0100
+++ libsigrok-0.4.0/configure.ac	2016-07-02 08:53:49.128940025 +0200
@@ -506,7 +506,7 @@
 AM_COND_IF([BINDINGS_CXX], [
 	PKG_CHECK_MODULES([LIBSIGROKCXX], [$SR_PKGLIBS_CXX])
 	PKG_CHECK_MODULES([PYSIGROK], [$SR_PKGLIBS_PYTHON $SR_PKGLIBS_CXX])
-	PKG_CHECK_MODULES([RBSIGROK], [$SR_PKGLIBS_RUBY $SR_PKGLIBS_CXX], [AC_SUBST(RBSIGROK_EXTDIR, "lib/$($PKG_CONFIG --variable=sitearch $SR_PKGLIBS_RUBY)/$($PKG_CONFIG --variable=RUBY_BASE_NAME $SR_PKGLIBS_RUBY)/vendor_ruby/$($PKG_CONFIG --variable=ruby_version $SR_PKGLIBS_RUBY)")])
+	PKG_CHECK_MODULES([RBSIGROK], [$SR_PKGLIBS_RUBY $SR_PKGLIBS_CXX], [AC_SUBST(RBSIGROK_EXTDIR, "$($PKG_CONFIG --variable=vendorarchdir $SR_PKGLIBS_RUBY)")])
 ])
 
 # Check for specific libusb features, now that we know the CFLAGS.
--- libsigrok-0.5.0/Makefile.am~	2018-07-04 20:24:21.000000000 +0200
+++ libsigrok-0.5.0/Makefile.am	2018-07-04 20:30:43.023460726 +0200
@@ -757,8 +757,8 @@ $(REXT): $(ROBJ) @ORDER@ bindings/cxx/li
 ruby-build: $(REXT)
 
 ruby-install: $(REXT)
-	$(INSTALL) -d $(DESTDIR)$(prefix)/$(RBSIGROK_EXTDIR)
-	$(INSTALL) $(REXT) $(DESTDIR)$(prefix)/$(RBSIGROK_EXTDIR)
+	$(INSTALL) -d $(DESTDIR)$(RBSIGROK_EXTDIR)
+	$(INSTALL) $(REXT) $(DESTDIR)$(RBSIGROK_EXTDIR)
 
 ruby-uninstall:
 	rm -f $(DESTDIR)$(prefix)/$(RBSIGROK_EXTDIR)/sigrok.$(RUBY_DLEXT)
