diff -urNp -x '*.orig' libsigrok-0.5.2.org/Makefile.am libsigrok-0.5.2/Makefile.am
--- libsigrok-0.5.2.org/Makefile.am	2019-12-25 21:29:19.000000000 +0100
+++ libsigrok-0.5.2/Makefile.am	2021-03-05 22:45:56.574087898 +0100
@@ -822,7 +822,7 @@ $(PDIR)/timestamp: $(PDIR)/sigrok/core/c
 python-install:
 	$(AM_V_at)$(MKDIR_P) "$(DESTDIR)$(prefix)" "$(DESTDIR)$(exec_prefix)"
 	destdir='$(DESTDIR)'; $(setup_py) install $${destdir:+"--root=$$destdir"} \
-		--prefix "$(prefix)" --exec-prefix "$(exec_prefix)"
+		--prefix "$(prefix)" --exec-prefix "$(exec_prefix)" --install-purelib="$(pythondir)" --optimize=2
 
 python-clean:
 	-$(AM_V_at)rm -f $(PDIR)/timestamp
--- libsigrok-0.5.2/bindings/python/sigrok/core/classes.i.orig	2019-12-25 21:29:19.000000000 +0100
+++ libsigrok-0.5.2/bindings/python/sigrok/core/classes.i	2025-04-06 15:44:05.193989617 +0200
@@ -131,7 +131,7 @@ typedef guint pyg_flags_type;
 
         auto arglist = Py_BuildValue("(OO)", log_obj, string_obj);
 
-        auto result = PyEval_CallObject($input, arglist);
+        auto result = PyObject_CallObject($input, arglist);
 
         Py_XDECREF(arglist);
         Py_XDECREF(log_obj);
@@ -174,7 +174,7 @@ typedef guint pyg_flags_type;
     $1 = [=] () {
         const auto gstate = PyGILState_Ensure();
 
-        const auto result = PyEval_CallObject($input, nullptr);
+        const auto result = PyObject_CallObject($input, nullptr);
         const bool completed = !PyErr_Occurred();
         const bool valid_result = (completed && result == Py_None);
 
@@ -218,7 +218,7 @@ typedef guint pyg_flags_type;
 
         auto arglist = Py_BuildValue("(OO)", device_obj, packet_obj);
 
-        auto result = PyEval_CallObject($input, arglist);
+        auto result = PyObject_CallObject($input, arglist);
 
         Py_XDECREF(arglist);
         Py_XDECREF(device_obj);
