diff -urNp -x '*.orig' libsigrok-0.5.2.org/Makefile.am libsigrok-0.5.2/Makefile.am
--- libsigrok-0.5.2.org/Makefile.am	2021-03-05 22:45:56.830753458 +0100
+++ libsigrok-0.5.2/Makefile.am	2021-03-05 22:45:56.894086518 +0100
@@ -932,9 +932,11 @@ $(JLIB): $(JCXX) \
 		-fno-strict-aliasing -fPIC -shared $(JCLS)/classes_wrap.cxx \
 		-lsigrokcxx $(LIBSIGROKCXX_LIBS) $(SR_EXTRA_CXX_LIBS) -o $(JLIB)
 
+jnilibdir = $(libdir)
+
 java-install:
-	$(INSTALL) -d $(DESTDIR)$(libdir)/jni
-	$(INSTALL) $(JLIB) $(DESTDIR)$(libdir)/jni
+	$(INSTALL) -d $(DESTDIR)$(jnilibdir)
+	$(INSTALL) $(JLIB) $(DESTDIR)$(jnilibdir)
 	$(INSTALL) -d $(DESTDIR)$(datadir)/java
 	$(INSTALL) $(JJAR) $(DESTDIR)$(datadir)/java
 
