--- libsigrok-0.4.0/Makefile.am.orig	2016-06-30 21:35:37.630787111 +0200
+++ libsigrok-0.4.0/Makefile.am	2016-07-01 05:59:31.576995763 +0200
@@ -675,16 +675,16 @@
 		$(nodist_library_include_HEADERS) \
 		$(bindings_cxx_libsigrokcxx_la_include_HEADERS) \
 		$(nodist_bindings_cxx_libsigrokcxx_la_include_HEADERS)
-	$(AM_V_CXX)$(CXX) $(RBSIGROK_CFLAGS) -I. -Iinclude -Ibindings/cxx/include -fPIC -o $@ -c $<
+	$(AM_V_CXX)$(CXX) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_CPPFLAGS) $(CPPFLAGS) $(RBSIGROK_CFLAGS) -I. -Iinclude -Ibindings/cxx/include -fPIC -o $@ -c $<
 
 $(REXT): $(ROBJ) @ORDER@ bindings/cxx/libsigrokcxx.la
-	$(AM_V_CXXLD)$(CXX) -shared -std=c++11 -o $@ $< -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
+	$(AM_V_CXXLD)$(CXX) $(AM_CXXFLAGS) $(CXXFLAGS) -shared -std=c++11 -o $@ $< -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
 
 ruby-build: $(REXT)
 
 ruby-install: $(REXT)
-	$(INSTALL) -d $(DESTDIR)$(prefix)/$(RBSIGROK_EXTDIR)
-	$(INSTALL) $< $(DESTDIR)$(prefix)/$(RBSIGROK_EXTDIR)
+	$(INSTALL) -d $(DESTDIR)$(RBSIGROK_EXTDIR)
+	$(INSTALL) $< $(DESTDIR)$(RBSIGROK_EXTDIR)
 
 ruby-clean:
 	-$(AM_V_at)rm -fr $(RDIR)/doc
--- libsigrok-0.4.0/configure.ac.orig	2016-01-29 23:44:34.000000000 +0100
+++ libsigrok-0.4.0/configure.ac	2016-07-02 08:53:49.128940025 +0200
@@ -506,7 +506,7 @@
 AM_COND_IF([BINDINGS_CXX], [
 	PKG_CHECK_MODULES([LIBSIGROKCXX], [$SR_PKGLIBS_CXX])
 	PKG_CHECK_MODULES([PYSIGROK], [$SR_PKGLIBS_PYTHON $SR_PKGLIBS_CXX])
-	PKG_CHECK_MODULES([RBSIGROK], [$SR_PKGLIBS_RUBY $SR_PKGLIBS_CXX], [AC_SUBST(RBSIGROK_EXTDIR, "lib/$($PKG_CONFIG --variable=sitearch $SR_PKGLIBS_RUBY)/$($PKG_CONFIG --variable=RUBY_BASE_NAME $SR_PKGLIBS_RUBY)/vendor_ruby/$($PKG_CONFIG --variable=ruby_version $SR_PKGLIBS_RUBY)")])
+	PKG_CHECK_MODULES([RBSIGROK], [$SR_PKGLIBS_RUBY $SR_PKGLIBS_CXX], [AC_SUBST(RBSIGROK_EXTDIR, "$($PKG_CONFIG --variable=vendorarchdir $SR_PKGLIBS_RUBY)")])
 ])
 
 # Check for specific libusb features, now that we know the CFLAGS.
